rcl_yaml_param_parser/test/test_parse_yaml.cpp

```diff
@@ -48,47 +48,57 @@ TEST(test_parser, correct_syntax) {
   bool res = rcl_parse_yaml_file(path, params_hdl);
   ASSERT_TRUE(res) << rcutils_get_error_string().str;
 
-  rcl_variant_t * param_value = rcl_yaml_node_struct_get("lidar_ns/lidar_1", "ports", params_hdl);
-  ASSERT_TRUE(NULL != param_value) << rcutils_get_error_string().str;
-  ASSERT_TRUE(NULL != param_value->integer_array_value);
-  ASSERT_EQ(3U, param_value->integer_array_value->size);
-  EXPECT_EQ(2438, param_value->integer_array_value->values[0]);
-  EXPECT_EQ(2439, param_value->integer_array_value->values[1]);
-  EXPECT_EQ(2440, param_value->integer_array_value->values[2]);
-  res = rcl_parse_yaml_value("lidar_ns/lidar_1", "ports", "[8080]", params_hdl);
-  EXPECT_TRUE(res) << rcutils_get_error_string().str;
-  ASSERT_TRUE(NULL != param_value->integer_array_value);
-  ASSERT_EQ(1U, param_value->integer_array_value->size);
-  EXPECT_EQ(8080, param_value->integer_array_value->values[0]);
+  rcl_params_t * copy_of_params_hdl = rcl_yaml_node_struct_copy(params_hdl);
+  ASSERT_TRUE(NULL != copy_of_params_hdl) << rcutils_get_error_string().str;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT({
+    rcl_yaml_node_struct_fini(copy_of_params_hdl);
+  });
 
-  param_value = rcl_yaml_node_struct_get("lidar_ns/lidar_2", "is_back", params_hdl);
-  ASSERT_TRUE(NULL != param_value) << rcutils_get_error_string().str;
-  ASSERT_TRUE(NULL != param_value->bool_value);
-  EXPECT_FALSE(*param_value->bool_value);
-  res = rcl_parse_yaml_value("lidar_ns/lidar_2", "is_back", "true", params_hdl);
-  EXPECT_TRUE(res) << rcutils_get_error_string().str;
-  ASSERT_TRUE(NULL != param_value->bool_value);
-  EXPECT_TRUE(*param_value->bool_value);
+  rcl_params_t * params_hdl_set[] = {params_hdl, copy_of_params_hdl};
+  for (rcl_params_t * params : params_hdl_set) {
+    rcl_variant_t * param_value =
+      rcl_yaml_node_struct_get("lidar_ns/lidar_1", "ports", params);
+    ASSERT_TRUE(NULL != param_value) << rcutils_get_error_string().str;
+    ASSERT_TRUE(NULL != param_value->integer_array_value);
+    ASSERT_EQ(3U, param_value->integer_array_value->size);
```

Comment:
We should test every possible param type. For example, this is not covering bool array.

Reply:
That predates this PR, but it is true nonetheless. See 37e15696f6cbd4cf1ff85950ddf63fc729f2271e.